crates/symbolicator-service/src/download/index.rs

```diff
@@ -325,28 +332,81 @@ impl SourceIndexService {
     /// This ID is stored in a file called `lastid.txt` in the
     /// `000Admin` directory.
     ///
-    /// The last ID is locally cached for an hour, so if a new
-    /// file is uploaded, we will see it at most an hour later.
+    /// The last ID is locally cached for an hour, and download failures
+    /// are cached for 10 minutes. If downloading a new last ID fails but there
+    /// is an existing succesful download, it will be reused for another hour.
     async fn fetch_symstore_last_id_memoized(
         &self,
         scope: Scope,
         source: &IndexSourceConfig,
     ) -> CacheContents<u32> {
-        self.symstore_last_id_cache
-            .get_with((scope, source.id().clone()), async {
-                let res = self.fetch_symstore_last_id(source).await;
-
-                if let Err(ref e) = res {
-                    tracing::error!(
-                        error = e as &dyn std::error::Error,
-                        source_id = %source.id(),
-                        "Failed to fetch Symstore index last ID",
-                    );
-                }
+        let compute = || async {
```

Comment:
Rust has async closures now 👀  (untested):
```suggestion
        let compute = async || {
```

Reply:
Tried it, it works.